components/script/dom/navigationpreloadmanager.rs

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::NavigationPreloadManagerBinding::NavigationPreloadState;
+use crate::dom::bindings::codegen::Bindings::NavigationPreloadManagerBinding::{
+    NavigationPreloadManagerMethods, Wrap,
+};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::ByteString;
+use crate::dom::domexception::{DOMErrorName, DOMException};
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::promise::Promise;
+use crate::dom::serviceworkerregistration::ServiceWorkerRegistration;
+use dom_struct::dom_struct;
+use js::jsval::UndefinedValue;
+use malloc_size_of::{MallocSizeOf, MallocSizeOfOps};
+use std::rc::Rc;
+
+impl MallocSizeOf for NavigationPreloadState {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.enabled.size_of(ops) + self.headerValue.size_of(ops)
+    }
+}
+
+#[dom_struct]
+pub struct NavigationPreloadManager {
+    reflector_: Reflector,
+    state: DomRefCell<NavigationPreloadState>,
```

Comment:
Per the specification, the ServiceWorkerRegistration should have `navigation preload enabled` and `navigation preload header value` members, rather than NavigationPreloadManager.

Reply:
Not sure if I read wrong spec, the `navigationPreload` field spec said it will return a `NavigationPreloadManager` (https://w3c.github.io/ServiceWorker/#service-worker-registration-navigationpreload)?

Reply:
Sorry, my comment was ambiguious. I meant that instead of NavigationPreloadManager storing values for "enabled" and "preload header", those members should be stored in ServiceWorkerRegistration instead.